src/slurm_plugin/clustermgtd.py

```diff
@@ -1124,38 +841,167 @@ def _handle_protected_mode_process(self, slurm_nodes, partitions, bootstrap_fail
         # Partition is not INACTIVE
         # Partition bootstrap failure count above threshold
         # Partition does not have job running
-        log.info(
-            "Partitions bootstrap failure count: %s, cluster will be set into protected mode if "
-            "protected failure count reach threshold",
-            self._partitions_protected_failure_count_map,
-        )
-        partitions_above_threshold = [
-            part
-            for part, failures in self._partitions_protected_failure_count_map.items()
-            if part not in self._inactive_partitions and failures >= self._config.protected_failure_count
-        ]
-        if partitions_above_threshold:
-            log.info("Bootstrap failure partitions: %s", print_with_count(partitions_above_threshold))
-            nodename_to_slurm_nodes_map = {node.name: node for node in slurm_nodes}
-            partitions_to_disable = ClusterManager._get_partitions_to_disable(
-                nodename_to_slurm_nodes_map, partitions_above_threshold, partitions
+        if self._partitions_protected_failure_count_map:
+            log.info(
+                "Partitions bootstrap failure count: %s, cluster will be set into protected mode if "
+                "protected failure counts reach threshold",
+                self._partitions_protected_failure_count_map,
             )
+
+        partitions_to_disable = []
+        bootstrap_failure_partitions_have_jobs = []
+        for part_name, failures in self._partitions_protected_failure_count_map.items():
+            part = partitions_name_map.get(part_name)
+            if part and not part.is_inactive() and failures >= self._config.protected_failure_count:
+                if part.has_running_job():
+                    bootstrap_failure_partitions_have_jobs.append(part_name)
+                else:
+                    partitions_to_disable.append(part_name)
+
+        if partitions_to_disable:
+            log.info("Bootstrap failure partitions: %s", partitions_to_disable)
+        if bootstrap_failure_partitions_have_jobs:
+            log.info(
+                "Bootstrap failure partitions %s currently have jobs running, not disabling them.",
+                bootstrap_failure_partitions_have_jobs,
+            )
+        if partitions_to_disable or bootstrap_failure_partitions_have_jobs:
```

Comment:
why not simple `if partitions_to_disable`. `_enter_protected_mode` should not be the one printing "Not entering protected mode since active jobs are running in bootstrap failure partitions".